### ISSUE TITLE
There are features that are not additive

### ISSUE THREAD

**Issue by JSorngard** [JASory/machine-prime#4]
Thanks for the super fast crate!

If you try to compile this crate with e.g. both the `small` and `tiny` features enabled the compilation fails.

This is a potential problem because if a user depends on two different crates that in turn both depend on `machine-prime` but use different features, that user will not be able to compile their program, as [cargo assumes that features are additive](https://doc.rust-lang.org/cargo/reference/features.html#feature-unification) and will only compile one version of the crate with both features enabled.   

This can also result in semver-compatible changes in dependencies becoming breaking changes. If crate A doesn't change its external API, and so only bumps the patch in its version number, but changes which features it enables in `machine-prime` from `small` to `tiny`, a user that uses crate A and crate B, where B depends on `machine-prime` with `small` will now see breaking changes when they run `cargo update`.

This may also just be a normal bug that wasn't caught, in which case this is just a normal bug report :P

**Comment by JSorngard** [comment]
The exact error in the above example is this:

![bild](https://github.com/user-attachments/assets/8c7bce62-fbee-4ad1-9fbd-2507619e051b)

**Comment by JSorngard** [comment]
This may also just be a bug that wasn't caught and the features are intended to be additive, in which case this issue is just a normal bug report :P

**Comment by JSorngard** [comment]
Tagging you @JASory to ensure github lets you know about this.

**Comment by JASory** [comment]
I was able to implement feature precedence, going from slower to faster. 
However, it will be about 2 weeks as I want to verify new parameters before I update the library.

**Comment by JASory** [comment]
It should be fixed now. I added some features and renamed the current ones. To get the same behaviour you need to add

```
version = "1.4.1"
default-features=false 
features = ["lucas"]
```

If you want to access the internal montgomery arithmetic, you can now use the "internal" feature. And the "wide" feature, promotes the functions to u128. It simply uses a Baille-PSW test so it's not known to have no pseudoprimes under 2^128 and may be incorrect.

**Comment by JSorngard** [comment]
Cool!

If a program depends on two crates and one crate relies on the functions taking u64, and the other crate enables the wide feature, will this cause a compilation failure?

**Comment by JASory** [comment]
Good point. Switching the function signature seems like a bad idea. I'll just add a is_prime_128 instead, that's how I'm calling it in my other libraries anyway.

**Comment by JASory** [comment]
The dependency conflicts should be fixed now.

**Comment by JSorngard** [comment]
Alright! I'll make an update soon to depend on this in `const-primes`.

**Comment by JSorngard** [comment]
I just have one additional thing: as far as I can tell the published version of the crate isn't `no_std` compatible, as the `#![no_std]` is commented out. Would it be possible to leave that in, such that the crate can be used in `no_std` environments?

**Comment by JASory** [comment]
I've never run it in a true no_std environment, so from my testing it seemed like adding ![no_std] in the calling library had the same effect. The crate now has the no_std enabled, since there is no reason not to. (Upon recollection, I may have originally written this on a version of Rust without stable no_std and never updated it).

**Comment by JSorngard** [comment]
Nice! Unfortunately it seems that the
```rust
#![allow(internal_features)]
#![feature(lang_items)]
```
parts were also uncommented, and now the crate can not be used without a nightly compiler.

**Comment by JASory** [comment]
Yanked it. The new version should work.

**Comment by JSorngard** [comment]
Now I've re-added the dependency on this crate in `const-primes` :D
